### ISSUE TITLE
Getting glyf errors with TrueType fonts in @font-face

### ISSUE THREAD

**Issue by gavin310** [dompdf/dompdf#3001]
I'm using TrueType fonts downloaded straight from Adobe (Proxima Nova) but I cannot get the fonts to work. First I was using domPDF v1 and only the Bold Italic font would give the error, but I upgraded to domPDF v2 and now it's every font.

I've tried converting the TTF files to OTF and then back to TTF, and also converting to WOFF2 and then back to TTF, hoping it would build the glyf table, but I still get the error. Is there maybe another reason these errors could be happening? Or is there a way it can be fixed?

```
Notice: Undefined index: glyf in /nas/content/live/abcsite/wp-content/themes/abctheme/includes/lib/vendor/phenx/php-font-lib/src/FontLib/TrueType/File.php on line 329

Fatal error: Uncaught Error: Call to a member function getGlyphIDs() on null in /nas/content/live/abcsite/wp-content/themes/abctheme/includes/lib/vendor/phenx/php-font-lib/src/FontLib/TrueType/File.php:202 Stack trace:
.0 /nas/content/live/abcsite/wp-content/themes/abctheme/includes/lib/vendor/dompdf/dompdf/lib/Cpdf.php(1191): FontLib\TrueType\File->setSubset(Array) 
.1 /nas/content/live/abcsite/wp-content/themes/abctheme/includes/lib/vendor/dompdf/dompdf/lib/Cpdf.php(951): Dompdf\Cpdf->processFont(8, Array) 
.2 /nas/content/live/abcsite/wp-content/themes/abctheme/includes/lib/vendor/dompdf/dompdf/lib/Cpdf.php(3215): Dompdf\Cpdf->o_font(8, 'add') 
.3 /nas/content/live/abcsite/wp-content/themes/abctheme/includes/lib/vendor/dompdf/dompdf/src/Adapter/CPDF.php(940): Dompdf\Cpdf->output(false) 
.4 /nas/content/live/abcsite/wp-content/themes/abctheme/includes/lib/vendor/dompdf/dompdf/src/Dompdf.php(895): Dompdf\Adapter\CPDF->output(Array) 
.5 /nas/content/live/abcsite/wp-content/plugins in /nas/content/live/abcsite/wp-content/themes/abctheme/includes/lib/vendor/phenx/php-font-lib/src/FontLib/TrueType/File.php on line 202
```

**Comment by gavin310** [comment]
Might as well include my @font-face declarations just in case:

```
@font-face {
	font-family: 'Proxima Nova';
	font-style: normal;
	font-weight: 300;
	src: url(../fonts/ProximaNova-Light.ttf) format("truetype");
}

@font-face {
	font-family: 'Proxima Nova';
	font-style: italic;
	font-weight: 300;
	src: url(../fonts/ProximaNova-LightItalic.ttf) format("truetype");
}

@font-face {
	font-family: 'Proxima Nova';
	font-style: normal;
	font-weight: normal;
	src: url(../fonts/ProximaNova-Regular.ttf) format("truetype");
}

@font-face {
	font-family: 'Proxima Nova';
	font-style: italic;
	font-weight: normal;
	src: url(../fonts/ProximaNova-Italic.ttf) format("truetype");
}

@font-face {
	font-family: 'Proxima Nova';
	font-style: normal;
	font-weight: bold;
	src: url(../fonts/ProximaNova-SemiBold.ttf) format("truetype");
}

@font-face {
	font-family: 'Proxima Nova';
	font-style: italic;
	font-weight: bold;
	src: url(../fonts/ProximaNova-SemiBoldItalic.ttf) format("truetype");
}
```

**Comment by bsweeney** [comment]
Also, it looks like the error is thrown when creating a font subset, so you _could_ disable font subsetting. Though that can significantly increase the site of the PDF if it's a large font.

Do you know what version of php-font-lib you're using? glyf table should be present it's probably just the library having trouble reading it.

After upgrading Dompdf you probably want to restore the font directory back to a vanilla state to avoid potential conflicts caused by a bad parse from a previous run.

**Comment by bsweeney** [comment]
Will probably close in favor of dompdf/php-font-lib#65, depending on how things go here.

**Comment by gavin310** [comment]
Thanks for the quick response!

Disabling subsetting isn't ideal, but I'm curious to know if that'll fix it. I can give that a try tomorrow as it's very late here.

The bower.json file says I'm using php-font-lib v0.3.1. When I installed domPDF v2 I first removed the old version with composer and then did a new `composer require dompdf/dompdf` so I'm pretty sure everything is in a vanilla state. I'll try again just to be sure.

No worries if you need to close this one.

**Comment by bsweeney** [comment]
Current version of php-font-lib is 0.5.4. I'm surprised it wasn't updated with Dompdf. Might try telling composer to update that library as well and see if the impacting this particular font was addressed.

**Comment by gavin310** [comment]
@bsweeney I actually did have the current version of php-font-lib, but the [bower.json](https://github.com/dompdf/php-font-lib/blob/master/bower.json) file in the current version still has 0.3.1 as the version and I assumed that was my version.

So disabling font subsetting worked! On the test PDF the original size was 2.2mb and the new size is 2.5mb. That's a significant increase, however for our purposes it's totally acceptable. THANK YOU!

For anyone else who wants to try to disable font subsetting, in the options I just set `$options->set( 'isFontSubsettingEnabled', false );`.
